### PR TITLE
mbedtls: move to external module

### PR DESCRIPTION
**Pull request by nashif**
Use external module from https://github.com/zephyrproject-rtos/mbedtls

Includes commits from other PRs.

### CODE REVIEW

**Comment by aescolar** [comment]
@nashif (it seems you are creating PRs for all ext/ components?)
(Given that the tinycbor trial was very easy to miss/ignore)
Would it make sense to start with a few first, give it a couple of weeks, and see if there is any trouble?
(maybe we will need to clarify something in the documentation or so)

File: ext/lib/crypto/CMakeLists.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 add_subdirectory_if_kconfig(tinycrypt)
-add_subdirectory_ifdef(CONFIG_MBEDTLS mbedtls)
+#add_subdirectory_ifdef(CONFIG_MBEDTLS mbedtls)
```
Comment (SebastianBoe) [inline ext/lib/crypto/CMakeLists.txt]: Is this a WIP?

Code should not be commented out.

**Comment by nashif** [comment]
> SebastianBoe changed the title mbedtls: move to external module move to external module a minute ago

Why?

**Comment by SebastianBoe** [comment]
Because it moves more than mbedtls to an external module.

**Comment by nashif** [comment]
> @nashif (it seems you are creating PRs for all ext/ components?)
> (Given that the tinycbor trial was very easy to miss/ignore)
> Would it make sense to start with a few first, give it a couple of weeks, and see if there is any trouble?
> (maybe we will need to clarify something in the documentation or so)

Waiting longer is just going to be a nightmare. Trying to keep up with PRs introducing changing to ext/ and keep this working is a full-time job. We want this to go in as fast as possible, deal with the issues and resolve them very early on. I do not expect this all to be merged today or tomorrow so it is going to take a while to process all of the modules anyways. Documentation and an announcement will be a prerequisite for those PRs to go in. Working on that.

**Comment by nashif** [comment]
> Because it moves more than mbedtls to an external module.

thats not the intention, I have multiple PRs, I include previous PRs just to keep things sane and in the right order, otherwise I will spend my day rebasing.

**Comment by SebastianBoe** [comment]
Ah, my mistake, I did not see

> Includes commits from other PRs.

**Comment by nashif** [comment]
@aescolar btw, there is no intention to remove everything in one PR, they will go in order and in a delayed fashion...

**Comment by nashif** [comment]
@SebastianBoe and a live demo: https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/41434/2/console

**Comment by aescolar** [comment]
@nashif As you prefer
( I did not intend to postpone it overall, was only proposing to start with maybe these 4 you already have PRs for now, which should affect already a big set of users, and expose any possible problems. Just in case there is something which would cause a lot of trouble, to not possibly affect everybody in one big bang. If say, in 2 weeks, things are calm, continue with all the rest)

**Comment by mbolivar** [comment]
@nashif needs rebase

File: ext/lib/crypto/Kconfig [hunk 2]
```diff
@@ -7,6 +7,5 @@
 menu "Cryptography"
 
 source "ext/lib/crypto/tinycrypt/Kconfig"
-source "ext/lib/crypto/mbedtls/Kconfig"
```
Comment (mbolivar) [inline ext/lib/crypto/Kconfig]: Do we need to keep this in the tree for dependency management? I don't understand the conclusion of that discussion about module kconfigs.

**Review by mbolivar — APPROVED** [review]
west changes lgtm

File: ext/lib/crypto/Kconfig (hunk 2 above)
Comment (nashif) [reply ext/lib/crypto/Kconfig]: it is in modules/Kconfig.mbedtls

**Pull request merged**